lazy-pages/src/host_func.rs

```diff
@@ -102,28 +102,31 @@ impl<'a> AccessHandler for HostFuncAccessHandler<'a> {
 fn accesses_pages(
     ctx: &mut LazyPagesExecutionContext,
     accesses: &[MemoryInterval],
-) -> Result<BTreeSet<GearPageNumber>, Error> {
-    let mut set = BTreeSet::new();
-    for access in accesses {
-        // TODO: here we suppose zero byte access like one byte access, because
-        // backend memory impl can access memory even in case access has size 0.
-        // We can optimize this if will ignore zero bytes access in core-backend (issue #2095).
-        let last_byte = access
-            .offset
-            .checked_add(access.size.saturating_sub(1))
-            .ok_or(Error::OutOfWasmMemoryAccess)?;
-
-        let page_size = GearPageNumber::size(ctx);
-        let mut offset = access.offset;
-        while offset <= last_byte {
-            set.insert(GearPageNumber::from_offset(ctx, offset));
-            offset = match offset.checked_add(page_size) {
-                Some(offset) => (offset / page_size) * page_size,
-                None => break,
+    pages: &mut BTreeSet<GearPageNumber>,
```

Comment:
I don't think return result thru `&mut` param is more effective way. Also it's less rust styled way)) so ...

Reply:
I don't think so, because this function will be 100% inlined, so this is not important to do things like this

Reply:
Why u need to create empty set inside fn?